pkg/advisory/create_test.go

```diff
@@ -203,8 +208,20 @@ func TestCreate(t *testing.T) {
 			}
 
 			if err == nil {
-				if diff := cmp.Diff(tt.expectedDoc, advisoryDocs.Select().WhereName(tt.req.Package).Configurations()[0]); diff != "" {
-					t.Errorf("Create() mismatch (-want +got):\n%s", diff)
+				diff := cmp.Diff(tt.expectedDoc, advisoryDocs.Select().WhereName(tt.req.Package).Configurations()[0], cmp.FilterPath(func(p cmp.Path) bool {
+					// Check if the path is accessing the ID field within the Advisories slice.
+					if len(p) < 2 {
+						return false
+					}
+					if p[len(p)-1].String() == ".ID" {
+						if _, ok := p[len(p)-2].(cmp.SliceIndex); ok {
+							return true
+						}
+					}
+					return false
```

Comment:
This seems a little obscure to me, maybe I just need to look at it for longer. What does that `if _, ok := p[len(p)-2].(cmp.SliceIndex); ok` part do exactly?

Reply:
that part was not needed, i got a little help with chatgpt on this, but seems it add things that were not needed 😅